{
  "errors": [],
  "results": [
    {
      "check_id": "rules.autofix.python.django.security.passwords.use-none-for-password-default.use-none-for-password-default",
      "end": {
        "col": 20,
        "line": 63
      },
      "extra": {
        "fix_regex": {
          "regex": "(password.*)\"\"",
          "replacement": "\\1None"
        },
        "is_ignored": false,
        "lines": "        new_password = request.data.get(\"password\", \"\")\n        validate_password(new_password, user=user)\n        user.set_password(new_password)\n        user.save()",
        "message": "'new_password' is using the empty string as its default and is being used to set\nthe password on 'user'. If you meant to set an unusable password, set\nthe default value to 'None' or call 'set_unusable_password()'.\n",
        "metadata": {
          "cwe": "CWE-521: Weak Password Requirements",
          "owasp": "A2: Broken Authentication",
          "references": [
            "https://docs.djangoproject.com/en/3.0/ref/contrib/auth/#django.contrib.auth.models.User.set_password"
          ]
        },
        "metavars": {
          "$MODEL": {
            "abstract_content": "user",
            "end": {
              "col": 13,
              "line": 62,
              "offset": 2380
            },
            "start": {
              "col": 9,
              "line": 62,
              "offset": 2376
            },
            "unique_id": {
              "sid": 18,
              "type": "id"
            }
          },
          "$VAR": {
            "abstract_content": "new_password",
            "end": {
              "col": 21,
              "line": 60,
              "offset": 2281
            },
            "start": {
              "col": 9,
              "line": 60,
              "offset": 2269
            },
            "unique_id": {
              "sid": 20,
              "type": "id"
            }
          },
          "$W": {
            "abstract_content": "data",
            "end": {
              "col": 36,
              "line": 60,
              "offset": 2296
            },
            "start": {
              "col": 32,
              "line": 60,
              "offset": 2292
            },
            "unique_id": {
              "md5sum": "<masked in tests>",
              "type": "AST"
            }
          },
          "$X": {
            "abstract_content": "\"password\"",
            "end": {
              "col": 51,
              "line": 60,
              "offset": 2311
            },
            "start": {
              "col": 41,
              "line": 60,
              "offset": 2301
            },
            "unique_id": {
              "md5sum": "<masked in tests>",
              "type": "AST"
            }
          }
        },
        "severity": "ERROR"
      },
      "path": "targets/autofix/django-none-password-default.py",
      "start": {
        "col": 9,
        "line": 60
      }
    },
    {
      "check_id": "rules.autofix.python.django.security.passwords.use-none-for-password-default.use-none-for-password-default",
      "end": {
        "col": 20,
        "line": 81
      },
      "extra": {
        "fix_regex": {
          "regex": "(password.*)\"\"",
          "replacement": "\\1None"
        },
        "is_ignored": false,
        "lines": "    def create_user(self, email, password=\"\"):\n        \"\"\"\n        Creates and saves a Poster with the given email and password.\n        \"\"\"\n        if not email:\n            raise ValueError('Users must have an email address')\n\n        user = self.model(email=self.normalize_email(email))\n        user.set_password(password)\n        user.save(using=self._db)\n        return user",
        "message": "'password' is using the empty string as its default and is being used to set\nthe password on 'user'. If you meant to set an unusable password, set\nthe default value to 'None' or call 'set_unusable_password()'.\n",
        "metadata": {
          "cwe": "CWE-521: Weak Password Requirements",
          "owasp": "A2: Broken Authentication",
          "references": [
            "https://docs.djangoproject.com/en/3.0/ref/contrib/auth/#django.contrib.auth.models.User.set_password"
          ]
        },
        "metavars": {
          "$F": {
            "abstract_content": "create_user",
            "end": {
              "col": 20,
              "line": 71,
              "offset": 2698
            },
            "start": {
              "col": 9,
              "line": 71,
              "offset": 2687
            },
            "unique_id": {
              "md5sum": "<masked in tests>",
              "type": "AST"
            }
          },
          "$MODEL": {
            "abstract_content": "user",
            "end": {
              "col": 13,
              "line": 79,
              "offset": 2981
            },
            "start": {
              "col": 9,
              "line": 79,
              "offset": 2977
            },
            "unique_id": {
              "sid": 25,
              "type": "id"
            }
          },
          "$VAR": {
            "abstract_content": "password",
            "end": {
              "col": 42,
              "line": 71,
              "offset": 2720
            },
            "start": {
              "col": 34,
              "line": 71,
              "offset": 2712
            },
            "unique_id": {
              "sid": 24,
              "type": "id"
            }
          }
        },
        "severity": "ERROR"
      },
      "path": "targets/autofix/django-none-password-default.py",
      "start": {
        "col": 5,
        "line": 71
      }
    }
  ]
}